src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExampleFileTest.java

```diff
@@ -0,0 +1,175 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2025 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import java.beans.PropertyDescriptor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.internal.utils.XdocUtil;
+
+public class XdocsExampleFileTest {
+
+    private static final Set<String> IGNORE_PROPERTIES = Set.of(
+            "severity", "id", "fileExtensions", "tabWidth",
+            "fileContents", "tokens", "javadocTokens"
+    );
+
+    private static final Map<String, Set<String>> SUPPRESSED_PROPERTIES_BY_CHECK = Map.ofEntries(
+            Map.entry("JavadocMetadataScraper",
+                    Set.of("violateExecutionOnNonTightHtml", "writeXmlOutput")),
+            Map.entry("IllegalTokenTextCheck",
+                    Set.of("message")),
+            Map.entry("ImportOrderCheck",
+                    Set.of(
+                            "ordered",
```

Comment:
https://checkstyle.org/checks/imports/importorder.html#Example2-config
We have all .